sklearn/feature_extraction/image.py

```diff
@@ -139,6 +138,21 @@ def _to_graph(
     return return_as(graph)
 
 
+@validate_params(
+    {
+        "img": ["array-like"],  # Could it also be "sparse matrix"?
+        "mask": [None, "array-like"],
+        "return_as": [type(np.ndarray), type(sparse.spmatrix)],
+        "dtype": [
+            None,
+            type(float),
+            type(int),
+            type(bool),
+            type(str),  # for str valued dtypes
+            type(type),  # Any type is allowed by numpy DTypeLike at the moment.
```

Comment:
I don't think we will need all of those for the moment. For estimator, we delegate the validation to NumPy or SciPy which will raise an error if the `dtype` is not right.

E.g.:

https://github.com/scikit-learn/scikit-learn/blob/64432e1cf65fc87c763737aa6422bb92d3573786/sklearn/preprocessing/_encoders.py#L444-L457

Here we are exactly in this setting so I don't think that we need to make it more complex.